### PR TITLE
Add changes to ./configure to build on FreeBSD

### PR DESCRIPTION
**Pull request by jyn514**
See https://github.com/oilshell/oil/pull/411#issuecomment-509020340 and https://github.com/oilshell/oil/issues/415#issuecomment-509854207 for discussion.

### CODE REVIEW

**Comment by jyn514** [comment]
I'll do one better, this builds on Android as well :)

File: configure [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # Other settings: LTO, PGO?  Consider moving prefix, LTO, PGO to build and
 # install steps.
 
-TMP=${TMP:-/tmp}  # Assume that any system has $TMP set or /tmp exists
+TMP=${TMPDIR:-/tmp}  # Assume that any system has $TMP set or /tmp exists
```
Comment (andychu) [inline configure]: The comment is now out of date.  Does FreeBSD not have either TMP or /tmp ?

File: configure [hunk 2]
```diff
@@ -66,7 +66,7 @@ while true; do
       ;;
 
     --readline=*)
-      FLAG_readline=$(expr "$1" : '--readline=\(.*\)')
+      FLAG_readline=$(expr \( "$1" : '--readline=\(.*\)' \) )
```
Comment (andychu) [inline configure]: Please add a comment saying that `\(` is necessary for FreeBSD (which is honestly a great example of sucky language design in shell tools).

And link to the manual  too:

https://www.freebsd.org/cgi/man.cgi?query=expr&sektion=&n=1

File: configure (hunk 1 above)
Comment (jyn514) [reply configure]: Android does not have /tmp or $TMP, but it has $TMPDIR. FreeBSD has /tmp but not $TMP or $TMPDIr.

**Comment by jyn514** [comment]
I'm not using the NDK, I have [termux](https://github.com/termux/termux-app) installed. It has a mostly POSIX environment, minus a few quirks like missing /tmp and needing -landroid-glob for `glob` and `globfree`.

File: configure [hunk 3]
```diff
@@ -12,7 +12,7 @@
 # Other settings: LTO, PGO?  Consider moving prefix, LTO, PGO to build and
 # install steps.
 
-TMP=${TMPDIR:-/tmp}  # Assume that any system has $TMP set or /tmp exists
+TMP=${TMPDIR:-/tmp}  # Assume that any system has $TMPDIR set or /tmp exists
```
Comment (andychu) [inline configure]: Hm I was wondering why you removed TMP.  But this answer seems to say that TMPDIR or /tmp should exist.

https://unix.stackexchange.com/questions/174817/finding-the-correct-tmp-dir-on-multiple-platforms

Comment (andychu) [reply configure]: Ah OK I see you mentioned that in the commit description.  Thanks.

**Pull request merged**